sunpy/map/tests/test_mapbase.py

```diff
@@ -194,8 +203,7 @@ def test_detector(generic_map):
 
 
 def test_dsun(generic_map):
-    with pytest.warns(SunpyUserWarning, match='Missing metadata for observer: assuming Earth-based observer.*'):
```

Comment:
Removing the need to catch this everywhere is certainly a good idea, but do we still have a test somewhere that makes sure this warning is raised when the observer metadata is missing?